_posts/2019-08-27-bazel-0.29.0.md

```diff
@@ -0,0 +1,56 @@
+
+Following our [backward compatibility policy](https://docs.bazel.build/versions/master/backward-compatibility.html), Bazel 1.0 will introduce some [breaking changes](https://github.com/bazelbuild/bazel/labels/breaking-change-1.0). Please follow  "[Updating Bazel](https://docs.bazel.build/versions/master/updating-bazel.html)" guide to prepare your project for those changes.
+
+Please [report](https://github.com/bazelbuild/bazel/issues/new) any update problems.
+
+## Starlark
+
+*   [`repository_ctx.read`](https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#read) can now read files from anywhere on the system, and not just within the repository rule context.
+*   A new incompatible flag [`--incompatible_disallow_unverified_http_download`](https://github.com/bazelbuild/bazel/issues/8607) has been added that disallows downloads via `http` (as opposed to `https`) unless a checksum is given. The flag will be flipped in Bazel 1.0 (but remain present throughout the 1.x cycle). To migrate, add a known-good checksum or at least one `https` source for downloading the file.
+*   A new incompatible flag [`--incompatible_disable_target_provider_fields`](https://github.com/bazelbuild/bazel/issues/9014) removes the ability to access a target's providers via the field syntax (for example, `ctx.attr.dep.my_provider`). The provider-key syntax should be used instead (for example, `ctx.attr.dep[MyProvider]`).
+
+
+## Execution
+
+*   Added two new tags to better control remote caching/execution: `no-remote-cache` and `no-remote-exec`. Read the [docs](https://docs.bazel.build/versions/master/be/common-definitions.html#common-attributes) for details.
+*   The [docker sandbox](https://docs.bazel.build/versions/master/remote-execution-sandbox.html) and the [execution log](https://docs.bazel.build/versions/master/remote-execution-caching-debug.html) now respect `--remote_default_platform_properties`.
+*   The [platform](https://docs.bazel.build/versions/master/be/platform.html#platform) attribute `remote_execution_properties` has been deprecated in favor of the newly introduced `exec_properties` attribute.
+
+
+## Java
+
+*   We fixed remote execution on Windows for the [Java rules](https://docs.bazel.build/versions/master/be/java.html). See [#8614](https://github.com/bazelbuild/bazel/issues/8614).
+*   You can now use a new [`java_toolchain`](https://docs.bazel.build/versions/master/be/java.html#java_toolchain) target `@remote_java_tools_{linux/windows/darwin}//:prebuilt_toolchain` in your builds. The new toolchain uses only pre-built tools, including singlejar and ijar, even on remote execution. The prebuilt toolchain  should be used only when host and execution platform are the same, otherwise the binaries will not work on the execution platform.
```

Comment:
> @remote_java_tools_{linux/windows/darwin}//:prebuilt_toolchain

This is neither valid Bash nor valid Label syntax.
Could you write maybe `@remote_java_tools_<PLATFORM>//:prebuilt_toolchain` and specify the values for that?

Like so:
> `@remote_java_tools_<PLATFORM>//:prebuilt_toolchain` (where `<PLATFORM>` is one of `linux`, `windows`, or `darwin`